docs/guides/integration.md

```diff
@@ -0,0 +1,423 @@
+# Integration Testing With Earthly
+
+## Why a workflow for integration tests is important
+Running unit tests in a build pipeline is relatively simple.  By definition, unit tests have no external dependencies.  Things get more interesting when we want to test how our service integrates with other services and external systems.  A service may have dependencies on external file systems, on databases, on external message queues, or other services. An ergonomic and effective development environment should have simple ways to construct and run integration tests.  It should be easy to run these tests locally on the developer machine and in the build pipeline.  
+
+## A Scenario
+
+Imagine this scenario: You work on a team that owns a service, that, in production runs containerized in a microservice environment.  The totality of the services that make-up production are too numerous reasonably run on a developer machine but thankfully your service only depends on a handful of other services.  You have unit tests but need ways to test your points of integration with your dependencies.  You also want ways to test your service externally, that is you want to exercise it in ways a service consumer might. 
```

Comment:
`You also want ways to test your service externally, that is you want to exercise it in ways a service consumer might.`

I think we can use the term "end-to-end test" here.

Reply:
yeah, I think you are right.  I struggle here with the names of the testing approaches.  I think it could be argued that end to end testing would involve all the services that work together, not just this service and its dependencies, but not sure that distinction matters here.